spec/policy/token_introspection/token_introspection_spec.lua

```diff
@@ -0,0 +1,94 @@
+local test_backend_client = require('resty.http_ng.backend.test')
+local cjson = require('cjson')
+describe("token introspection policy", function()
+  describe("execute introspection", function()
+    local context
+    local test_backend
+
+    before_each(function()
+      test_backend = test_backend_client.new()
+      ngx.var = {}
+      ngx.var.http_authorization = "Bearer test"
+      context = {
+        service = {
+          auth_failed_status = 403,
+          error_auth_failed = "auth failed"
+        }
+      }
+    end)
+
+    it('execute token introspection success', function()
+      local introspection_url = "http://example/token/introspection"
+      local policy_config = {
+        client = test_backend,
+        introspection_url = introspection_url,
+        client_id = "client",
+        client_secret = "secret"
+      }
+
+      test_backend.expect{ url = introspection_url }.
```

Comment:
It probably should expect right http method  (POST)  too ? And some other parameters like token_type_hint ? And possibly the headers?